### PR TITLE
Remove user.data() in firestore data set

### LINKED ISSUE #2261: Remove fetching and pushing of currently present data of user in addOrUpdate
### Issue Description

Currently, In `models/users.js`, we are fetching current data of user and pushing it while updating profileDiffs, but as ` { merge : true } ` is used, this has no effect and no use, hence should be removed.

### Expected Behavior

`user.data()` is not fetched or set, while setting profileDiffs

### Current Behavior

`user.data()` is being fetched and set while updating profileDiffs.

### Screenshots

image

### Reproducibility

 - [x] This issue is reproducible
 - [ ] This issue is not reproducible

### Steps to Reproduce
1. 
2. 

### Severity/Priority
 - [ ] Critical
 - [ ] High
 - [ ] Medium
 - [x] Low

### Checklist

- [x] I have read and followed the project's code of conduct.
- [x] I have searched for similar issues before creating this one.
- [x] I have provided all the necessary information to understand and reproduce the issue.
- [x] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by RahulGoyal-tech**
Date: 03 12 2024

Developer Name: Rahul Goyal

---

## Issue Ticket Number

- Closes #2261 
## Description
- Removed user.data() in firestore set as we are already doing merge : true

### Documentation Updated?

- [x] Yes
- [ ] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

### CODE REVIEW

**Review by Achintya-Chatterjee — APPROVED** [review]

**Review by Achintya-Chatterjee — DISMISSED** [review]

**Review by iamitprakash — CHANGES_REQUESTED** [review]
where is Feature Flag ..?

**Comment by RahulGoyal-tech** [comment]
Added dev feature flag

**Comment by prakashchoudhary07** [comment]
No tests for it?

File: controllers/users.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ const verifyUser = async (req, res) => {
     if (!req.userData?.profileURL) {
       return res.boom.serverUnavailable("ProfileURL is Missing");
     }
-    await userQuery.addOrUpdate({ profileStatus: "PENDING" }, userId);
+    await userQuery.addOrUpdate({ profileStatus: "PENDING" }, userId, req.query.dev);
```
Comment (prakashchoudhary07) [inline controllers/users.js:46]: Please create a separate variable for req.query.dev and pass it here

File: controllers/users.js [hunk 2]
```diff
@@ -500,7 +500,7 @@ const updateSelf = async (req, res) => {
         const { roles } = discordMember;
         if (roles && roles.includes(discordDeveloperRoleId)) {
           if (req.body.disabledRoles && devFeatureFlag) {
-            const updatedUser = await userQuery.addOrUpdate({ disabled_roles: rolesToDisable }, userId);
+            const updatedUser = await userQuery.addOrUpdate({ disabled_roles: rolesToDisable }, userId, req.query.dev);
```
Comment (prakashchoudhary07) [inline controllers/users.js:503]: RE same as above

File: controllers/users.js [hunk 3]
```diff
@@ -514,7 +514,7 @@ const updateSelf = async (req, res) => {
       }
     }
 
-    const updatedUser = await userQuery.addOrUpdate(req.body, userId);
+    const updatedUser = await userQuery.addOrUpdate(req.body, userId, req.query.dev);
```
Comment (prakashchoudhary07) [inline controllers/users.js:517]: RE: same for similar things in this PR

**Comment by RahulGoyal-tech** [comment]
> No tests for it?

Behaviour remains same, only the code enhancement is done. Hence no tests.

File: controllers/users.js (hunk 1 above)
Comment (RahulGoyal-tech) [reply controllers/users.js]: resolved

File: controllers/users.js [hunk 4]
```diff
@@ -457,6 +458,7 @@ const updateSelf = async (req, res) => {
   try {
     const { id: userId, roles: userRoles, discordId } = req.userData;
     const devFeatureFlag = req.query.dev === "true";
+    const devFlag = req.query.dev;
```
Comment (prakashchoudhary07) [inline controllers/users.js:461]: Hey, so don't we already have a dev=true flag, for something else?

How are you making sure that your flag is not triggering the other code?

**Comment by prakashchoudhary07** [comment]
> > No tests for it?
> 
> Behaviour remains same, only the code enhancement is done. Hence no tests.

Please write tests for it

How is the new code that you added being tested?

File: models/users.js [hunk 5]
```diff
@@ -96,14 +96,24 @@ const addOrUpdate = async (userData, userId = null) => {
...
         }
-        await userModel.doc(userId).set(
-          {
-            ...user.data(),
-            ...userData,
-            updated_at: Date.now(),
-          },
-          { merge: true }
-        );
+        if (devFeatureFlag === "true") {
```
Comment (prakashchoudhary07) [inline models/users.js:99]: NIT: let devFeatureFlag be boolean

Comment (RahulGoyal-tech) [reply models/users.js:99]: Alright!! Will update as per that.

**Review by Achintya-Chatterjee — COMMENTED** [review]
tests are failing, kindly look into this

File: controllers/users.js (hunk 4 above)
Comment (RahulGoyal-tech) [reply controllers/users.js]: I changed the logic to boolean, hence using the same flag and not making a new one.

**Comment by RahulGoyal-tech** [comment]
> tests are failing, kindly look into this

Can you please once re run the tests. This has happened for me in the last PR where they failed due to some error.

**Review by Achintya-Chatterjee — APPROVED** [review]

**Review by prakashchoudhary07 — APPROVED** [review]

**Pull request merged**